src/classes/mainSyncProcess.ts

```diff
@@ -121,16 +121,22 @@ export class MainSyncProcess {
 
       // Otherwise, record was modified or we don't have it yet, so we'll store it
       // We also want to see the difference between the two
-      const objectDifference = jsonDiff.diff(
-        JSON.parse(storedTextEntry || "{}"),
-        JSON.parse(tmsTextEntry),
-        { raw: true }
-      );
-      Logger.debug(
-        `Difference in JSON exists between Stored TextEntry and TMS TextEntry for ${tmsRecord.ID}`
-      );
-      DiffLogger.debug({ [tmsRecord.ID]: objectDifference });
-      acc.push({ ID: tmsRecord.ID, TextEntry: tmsRecord.TextEntry });
+      try {
+        const objectDifference = jsonDiff.diff(
+          JSON.parse(storedTextEntry || "{}"),
+          JSON.parse(tmsTextEntry),
+          { raw: true }
+        );
+        Logger.debug(
+          `Difference in JSON exists between Stored TextEntry and TMS TextEntry for ${tmsRecord.ID}`
+        );
+        DiffLogger.debug({ [tmsRecord.ID]: objectDifference });
+        acc.push({ ID: tmsRecord.ID, TextEntry: tmsRecord.TextEntry });
+      } catch (error) {
+        Logger.error(
+          `Error occurred generating difference in JSON between Stored TextEntry and TMS TextEntry for ${tmsRecord.ID}`
+        );
+      }
```

Comment:
Handles some of our fields in the Collection Payload from TMS - which sometimes have malformed JSON.